git_theta/updates/sparse.py

```diff
@@ -17,7 +19,18 @@ def name(self):
     async def calculate_update(
         self, parameter: Parameter, previous_parameter: Parameter
     ) -> Parameter:
-        return parameter - previous_parameter
+        diff = parameter - previous_parameter
+        update = scipy.sparse.csr_matrix(np.reshape(diff, (-1,)))
+        return {
+            "data": update.data,
+            "indices": update.indices,
+            "indptr": update.indptr,
+            "shape": parameter.shape,
+        }
 
     async def apply_update(self, update: Parameter, previous: Parameter) -> Parameter:
-        return update + previous
+        param_update = scipy.sparse.csr_matrix(
+            (update["data"], update["indices"], update["indptr"]),
+            shape=(1, np.prod(update["shape"])),
```

Comment:
Should `shape` be `(np.prod(update["shape"]), )` instead of `(1, np.prod(update["shape"]))`? In `calculate_update` the update is flattened to be a 1-d array so seems like `shape` should also be 1-d. Probably doesn't make a difference in the result if the tests are passing.

Reply:
The scipy sparse matrix always wants a 2d thing so it was erroring when the shape wasn't 2d.

We should add a comment here explaining that.

Reply:
Gotcha, sounds good

Reply:
In `calculate_update` method, I changed the reshape to (1, flattened_shape) so that it is consistent with the `apply_update` method.